version: '3.9'

services:
  db:
    image: postgres:15
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_DB=revline_db
      - POSTGRES_USER=revline_user
      - POSTGRES_PASSWORD=revline_pass
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U revline_user -d revline_db"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
     - app_network

  redis:
    image: redis:7-alpine
    ports:
      - "6379:6379"

  web:
    build:
      context: .
      dockerfile: Dockerfile
    command: python manage.py runserver 0.0.0.0:8000
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_started
    environment:
      - DEBUG=1
      - DATABASE_URL=postgresql://revline_user:revline_pass@db:5432/revline_db
      - REDIS_URL=redis://redis:6379/0
    networks:
     - app_network

volumes:
  postgres_data:

networks:
  app_network:
    driver: bridge
